### PR TITLE
fix: Add a step validate to the Issue New Card form

### PR DESCRIPTION
**Pull request by nkdengineer**
### Fixed Issues

$ https://github.com/Expensify/App/issues/50667
PROPOSAL: https://github.com/Expensify/App/issues/50667#issuecomment-2407910788

### Tests

1. Create a new workspace
2. Connect back account and enable Expensify Card
3. Go to Expensify Card page
4. Click New Issue card button
5. Do the steps to step confirmation
6. Click Issue card button
7. Verify that: Validate Code modal is displayed
8. Enter magic code and verify that: redirects to true page and validateCode param is sent
- [x] Verify that no errors appear in the JS console

### Offline tests

same as above
### QA Steps

same as above
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/6c4e8fd8-c25e-4bd4-b746-6fd755e92f5c

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/17b5b7e1-d1a2-4910-98dd-67d505771f9a

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/07fc3074-ffc0-4ec8-b4bf-17f9e691169d

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/0fc27b16-8f4d-4565-a91e-94133ff1ee03

</details>

### CODE REVIEW

**Comment by hungvu193** [comment]
Code wise LGTM, but let hold for https://github.com/Expensify/App/issues/50667#issuecomment-2417701346 a littie bit 😄

**Comment by hungvu193** [comment]
Let's merge main 😄 This one can be off hold.

**Comment by hungvu193** [comment]
Hey @nkdengineer, can you also handle the case when the user enter wrong validate code?

**Comment by nkdengineer** [comment]
@hungvu193 i'll update soon

**Comment by nkdengineer** [comment]
@hungvu193 as the solution mention [here](https://github.com/Expensify/App/issues/50667#issue-2581859818), we will send the `validateCode` with the `CreateAdminIssuedVirtualCard` API, and that API will be responsible for checking whether the `validateCode` is correct or not, right?

**Comment by hungvu193** [comment]
> @hungvu193 as the solution mention [here](https://github.com/Expensify/App/issues/50667#issue-2581859818), we will send the `validateCode` with the `CreateAdminIssuedVirtualCard` API, and that API will be responsible for checking whether the `validateCode` is correct or not, right?

Yes. If the API still return successful response with invalid code, then we will need BE changes

**Comment by hungvu193** [comment]
It looks like we still need to merge the FE's PR first. Let's add the generic error message to failure data for now and handle it from there 😄

**Comment by nkdengineer** [comment]
@hungvu193 i don't know what I should really do here. Can i do like in `NewContactMethodPage` component?

https://github.com/Expensify/App/blob/2db49d4759d0008e2fb177523cd574290513e3c8/src/pages/settings/Profile/Contacts/NewContactMethodPage.tsx#L165-L177

**Comment by nkdengineer** [comment]
@hungvu193 friendly bump here

**Comment by hungvu193** [comment]
You can also improve the loading logic of `ValidateCodeForm`, right now loading is depended on `account.loading` but it's incorrect in this case.

**Comment by hungvu193** [comment]
@nkdengineer do you have any questions?

**Comment by nkdengineer** [comment]
> You can also improve the loading logic of ValidateCodeForm, right now loading is depended on account.isLoading but it's incorrect in this case.

@hungvu193  how can I improve this logic in this case?

**Comment by hungvu193** [comment]
You can add a new props into `ValidateCodeActionModal` call isLoading.

Here's the example:
https://github.com/Expensify/App/pull/51135/files#diff-5a75ff8b317bf0b6472287731f466dbe6a71b34d2417aaac848a66ee88d397b4


In this PR, we can use the `isLoading` property from `ONYXKEYS.ISSUE_NEW_EXPENSIFY_CARD`

**Comment by nkdengineer** [comment]
@hungvu193 Thanks for the suggestion, I updated

File: src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx [hunk 1]
```diff
@@ -33,9 +37,14 @@ function ConfirmationStep({policyID, backTo}: ConfirmationStepProps) {
...
     const {isOffline} = useNetwork();
-
+    const [account] = useOnyx(ONYXKEYS.ACCOUNT);
     const [issueNewCard] = useOnyx(ONYXKEYS.ISSUE_NEW_EXPENSIFY_CARD);
-
+    const [pendingContactAction] = useOnyx(ONYXKEYS.PENDING_CONTACT_ACTION);
+    const contactMethod = UserUtils.getContactMethod();
+    const [loginList] = useOnyx(ONYXKEYS.LOGIN_LIST);
+    const loginData = loginList?.[pendingContactAction?.contactMethod ?? contactMethod];
+    const validateLoginError = ErrorUtils.getLatestErrorField(loginData, 'addedLogin');
```
Comment (hungvu193) [inline src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx:46]: This error should be from `ONYXKEYS.ISSUE_NEW_EXPENSIFY_CARD`.

File: src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx [hunk 2]
```diff
@@ -122,11 +131,33 @@ function ConfirmationStep({policyID, backTo}: ConfirmationStepProps) {
...
+                    if (!loginData) {
+                        return;
+                    }
+                    User.clearContactMethodErrors(addSMSDomainIfPhoneNumber(pendingContactAction?.contactMethod ?? contactMethod), 'addedLogin');
+                }}
+                onClose={() => {
+                    if (loginData?.errorFields && pendingContactAction?.contactMethod) {
+                        User.clearContactMethod(pendingContactAction?.contactMethod);
+                        User.clearUnvalidatedNewContactMethodAction();
+                    }
```
Comment (hungvu193) [inline src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx:154]: We should clear the error from `ONYXKEYS.ISSUE_NEW_EXPENSIFY_CARD` instead.

File: src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx [hunk 3]
```diff
@@ -122,11 +131,33 @@ function ConfirmationStep({policyID, backTo}: ConfirmationStepProps) {
...
+            <ValidateCodeActionModal
+                handleSubmitForm={submit}
+                isLoading={issueNewCard?.isLoading}
+                sendValidateCode={() => User.requestValidateCodeAction()}
+                validateError={validateLoginError}
+                clearError={() => {
+                    if (!loginData) {
+                        return;
+                    }
+                    User.clearContactMethodErrors(addSMSDomainIfPhoneNumber(pendingContactAction?.contactMethod ?? contactMethod), 'addedLogin');
```
Comment (hungvu193) [inline src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx:148]: We should clear the error from ONYXKEYS.ISSUE_NEW_EXPENSIFY_CARD instead.

File: src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx [hunk 4]
```diff
@@ -122,11 +131,33 @@ function ConfirmationStep({policyID, backTo}: ConfirmationStepProps) {
...
             </ScrollView>
+            <ValidateCodeActionModal
+                handleSubmitForm={submit}
+                isLoading={issueNewCard?.isLoading}
+                sendValidateCode={() => User.requestValidateCodeAction()}
+                validateError={validateLoginError}
+                clearError={() => {
+                    if (!loginData) {
+                        return;
+                    }
```
Comment (hungvu193) [inline src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx:147]:
```suggestion
```

**Review by hungvu193 — CHANGES_REQUESTED** [review]

**Comment by nkdengineer** [comment]
@hungvu193 i updated, please check again

File: src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx [hunk 5]
```diff
@@ -122,11 +129,30 @@ function ConfirmationStep({policyID, backTo}: ConfirmationStepProps) {
...
                 </View>
             </ScrollView>
+            <ValidateCodeActionModal
+                handleSubmitForm={submit}
+                isLoading={issueNewCard?.isLoading}
+                sendValidateCode={() => User.requestValidateCodeAction()}
+                validateError={validateLoginError}
+                clearError={() => {
+                    User.clearContactMethodErrors(addSMSDomainIfPhoneNumber(pendingContactAction?.contactMethod ?? contactMethod), 'addedLogin');
+                }}
```
Comment (hungvu193) [inline src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx:144]: Why do we need to `clearContactMethodErrors` here? 🤔 Doesn't it should be `Card.clearIssueNewCardError();`?

File: src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx [hunk 6]
```diff
@@ -122,11 +129,30 @@ function ConfirmationStep({policyID, backTo}: ConfirmationStepProps) {
...
+                onClose={() => {
+                    if (validateLoginError && pendingContactAction?.contactMethod) {
+                        User.clearContactMethod(pendingContactAction?.contactMethod);
```
Comment (hungvu193) [inline src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx:147]: Same here, why do we need `User.clearContactMethod(pendingContactAction?.contactMethod);` here?

**Review by hungvu193 — CHANGES_REQUESTED** [review]

File: src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx [hunk 7]
```diff
@@ -33,9 +37,12 @@ function ConfirmationStep({policyID, backTo}: ConfirmationStepProps) {
     const {translate} = useLocalize();
     const styles = useThemeStyles();
     const {isOffline} = useNetwork();
-
+    const [account] = useOnyx(ONYXKEYS.ACCOUNT);
     const [issueNewCard] = useOnyx(ONYXKEYS.ISSUE_NEW_EXPENSIFY_CARD);
-
+    const [pendingContactAction] = useOnyx(ONYXKEYS.PENDING_CONTACT_ACTION);
+    const contactMethod = UserUtils.getContactMethod();
+    const validateLoginError = issueNewCard?.errors;
```
Comment (hungvu193) [inline src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx:44]: AFAIR, Sometimes server return errors as an array, so let's use this instead:
```suggestion
    const validateLoginError = ErrorUtils.getLatestErrorMessageField(issueNewCard);
```

**Comment by nkdengineer** [comment]
I'll give an update soon

**Comment by nkdengineer** [comment]
thanks @hungvu193, i updated

File: src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx [hunk 8]
```diff
@@ -122,11 +126,29 @@ function ConfirmationStep({policyID, backTo}: ConfirmationStepProps) {
...
+            <ValidateCodeActionModal
+                handleSubmitForm={submit}
+                isLoading={issueNewCard?.isLoading}
+                sendValidateCode={() => User.requestValidateCodeAction()}
+                validateError={validateLoginError}
+                clearError={() => {
+                    Card.clearIssueNewCardError();
+                }}
+                onClose={() => {
+                    if (validateLoginError && pendingContactAction?.contactMethod) {
```
Comment (hungvu193) [inline src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx:143]: We dont even need this condition, our error isn't related to login error. Let's remove it as well

**Review by hungvu193 — CHANGES_REQUESTED** [review]

**Comment by hungvu193** [comment]
Cool. It's getting late here. I'll do another review tmr. Hopefully we can wrap it up this week!

**Comment by hungvu193** [comment]
I encountered this issue while testing this PR, when ValidateCodeActionModal shows up, if I press any number, the modal dismissed.

https://github.com/user-attachments/assets/dbaebe97-c98c-4a0e-86ec-14d0472c85d3


Also, you can test the failed case by enabling `Simulate failing network requests` and setting `MAX_REQUEST_RETRIES` to 1 (or 0).

**Comment by nkdengineer** [comment]
I'm investigating

**Comment by nkdengineer** [comment]
thanks @hungvu193, i fixed

**Comment by hungvu193** [comment]
I'll review again today

File: src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx [hunk 9]
```diff
@@ -122,11 +125,31 @@ function ConfirmationStep({policyID, backTo}: ConfirmationStepProps) {
                         isAlertVisible={!!errorMessage}
                         isDisabled={isOffline}
                         isLoading={issueNewCard?.isLoading}
-                        onSubmit={submit}
+                        onSubmit={() => setIsValidateCodeActionModalVisible(true)}
                         buttonText={translate('workspace.card.issueCard')}
                     />
                 </View>
             </ScrollView>
+            {issueNewCard && (
```
Comment (hungvu193) [inline src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx:133]:
```suggestion
            {!!issueNewCard && (
```

**Review by hungvu193 — CHANGES_REQUESTED** [review]

**Comment by hungvu193** [comment]
Found 2 issues:
1. Sometimes ValidateActionCodeModal got dismissed right after I pressed IssueCard

https://github.com/user-attachments/assets/4c0c4de7-5a9f-4ac5-901f-017759697bfe


2. `CreateExpensifyCard` was called successfully, however the modal was not dismissed and it's loading infinitely.

https://github.com/user-attachments/assets/e688bf17-ef82-44ce-8074-a0f1e44262ee

**Comment by nkdengineer** [comment]
> CreateExpensifyCard was called successfully, however the modal was not dismissed and it's loading infinitely.

This problem also occurs in the login form when we call the API and get the status 429 too many requests 


https://github.com/user-attachments/assets/7f9a3a8c-cd9f-4fd0-9809-d1517a264500

**Comment by nkdengineer** [comment]
> Sometimes ValidateActionCodeModal got dismissed right after I pressed IssueCard

I tried many times and couldn't reproduce it

**Comment by hungvu193** [comment]
> > CreateExpensifyCard was called successfully, however the modal was not dismissed and it's loading infinitely.
> 
> This problem also occurs in the login form when we call the API and get the status 429 too many requests
> 
>  bug-login.mov


tbh, I don't see any 429 error.

https://github.com/user-attachments/assets/449a427d-5210-4dcb-b268-d11586064feb

**Comment by hungvu193** [comment]
> > Sometimes ValidateActionCodeModal got dismissed right after I pressed IssueCard
> 
> I tried many times and couldn't reproduce it

I just realized that you need to reproduce the first bug (infinite loading) to reproduce this one, after dismissing the infinite validate code action modal, next time when you issue a new card, this bug will happen.

https://github.com/user-attachments/assets/f09283c0-0e21-43c5-930b-4e933de3bd54

**Comment by nkdengineer** [comment]
I'm investigating

**Comment by nkdengineer** [comment]
@hungvu193 after merging main i can't reproduce bug anymore, please help me check again

**Comment by hungvu193** [comment]
YEah I can't reproduce it anymore :/

**Comment by hungvu193** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



https://github.com/user-attachments/assets/8eef3218-629d-47e9-ae69-a0c0886e567f


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/user-attachments/assets/3045b47d-960e-426c-bdb4-2e85f2384f0d


</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

File: src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx [hunk 10]
```diff
@@ -122,11 +125,31 @@ function ConfirmationStep({policyID, backTo}: ConfirmationStepProps) {
...
+                        onSubmit={() => setIsValidateCodeActionModalVisible(true)}
                         buttonText={translate('workspace.card.issueCard')}
                     />
                 </View>
             </ScrollView>
+            {!!issueNewCard && (
+                <ValidateCodeActionModal
+                    handleSubmitForm={submit}
+                    isLoading={issueNewCard?.isLoading}
+                    sendValidateCode={() => User.requestValidateCodeAction()}
```
Comment (hungvu193) [inline src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx:137]: Let's add `hasMagicCodeBeenSent` props here to make it consistent with other places.

**Review by hungvu193 — CHANGES_REQUESTED** [review]

File: src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx (hunk 10 above)
Comment (nkdengineer) [reply src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx:139]: Screenshot 2024-11-13 at 15 08 49

@hungvu193 i updated

**Review by hungvu193 — APPROVED** [review]

**Review by tgolen — APPROVED** [review]

File: src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx [hunk 11]
```diff
@@ -122,11 +127,32 @@ function ConfirmationStep({policyID, backTo}: ConfirmationStepProps) {
...
+                    validateError={validateError}
+                    hasMagicCodeBeenSent={validateCodeSent}
```
Comment (Pujan92) [inline src/pages/workspace/expensifyCard/issueNew/ConfirmationStep.tsx:141]: We aren't not resetting the value of `validateCodeSent` which causes the issue https://github.com/Expensify/App/issues/52784

**Pull request merged**